.github/workflows/unit-tests-workflow.yml

```diff
@@ -0,0 +1,51 @@
+name: Unit tests workflow
+on:
+  push:
+    branches:
+      - "*"
```

Comment:
Nit: This will run on every branch for every commit. I would limit the scope to run only on PR's to main. That way PRs on feature branches aren't blocked due to dev code being pushed there.

Reply:
sure, sounds good!